### PR TITLE
PLAT-601 Defined ILocale, CurrentLocale, etc.

### PR DESCRIPTION
**Pull request by oliver-richers**

### CODE REVIEW

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+	public void validate() {
+
+		if (locale.getDecimalSeparator().isEmpty()) {
+			throw new SofticarUserException(CommonCoreI18n.THE_DECIMAL_SEPARATOR_MAY_NOT_BE_EMPTY);
+		}
+
+		if (locale.getDecimalSeparator().equals(locale.getDigitGroupSeparator())) {
+			throw new SofticarUserException(CommonCoreI18n.THE_DECIMAL_SEPARATOR_MUST_BE_DIFFERENT_FROM_THE_DIGIT_GROUP_SEPARATOR);
+		}
```
Comment (daniel-klose) [inline platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java:23]: I think there needs to be an additional test that the Seperators may only be a single symbol.
Also, if easily implemented, there should be a check that the separators may not be letters like `c` or numbers `5`

**Review by daniel-klose — CHANGES_REQUESTED** [review]

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java (hunk 1 above)
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java:23]: - hmm, does it have to be a single character? and why prevent letters?
- regardings digits, you are right, will add that

Comment (daniel-klose) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java:23]: well tbh, everything beside a single special character is pretty much useless as a seperator imo ¯\_(ツ)_/¯

Comment (alexander-schmidt) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java:23]: I think:
- Most characters might be valid separators - somewhere, for some use case. Even a space. Even emoji. Notable exceptions are **digits** and **control characters**. We _could_ enforce that those are not used - but it's probably not necessary as long as end-users cannot freely define individual locales. If an admin deliberately _wants_ to break the configuration by defining something dumb - let them.
- Allowing multi-character separators is likely to cause headache in all sorts of parsers/converters. We should avoid that unless absolutely necessary.

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/ILocale.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	 * Returns the {@link String} to be used as decimal separator.
+	 * <p>
+	 * Common values are the decimal comma (,) or decimal point (.).
+	 * <p>
+	 * This separator must be different to the separator returned by
+	 * {@link #getDigitGroupSeparator()}.
+	 *
+	 * @return the decimal separator (never <i>null</i> and not empty)
+	 */
+	String getDecimalSeparator();
```
Comment (alexander-schmidt) [inline platform-common-core/src/main/java/com/softicar/platform/common/core/locale/ILocale.java:20]: We could change this to return `Optional<Character>`. AFAIK, multi-character separators are not a thing.

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/ILocale.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	 * <p>
+	 * Common values are space ( ), comma (,), point (.) or apostrophe (') or
+	 * the empty {@link String}.
+	 * <p>
+	 * This separator must be different to the separator returned by
+	 * {@link #getDecimalSeparator()}.
+	 *
+	 * @return the decimal separator (never <i>null</i> but may be empty)
+	 */
+	String getDigitGroupSeparator();
```
Comment (alexander-schmidt) [inline platform-common-core/src/main/java/com/softicar/platform/common/core/locale/ILocale.java:33]: We could change this to return `Optional<Character>`. AFAIK, multi-character separators are not a thing.

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/ILocale.java (hunk 2 above)
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/locale/ILocale.java:20]: I had this previously, but I prefer String here.

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/ILocale.java (hunk 3 above)
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/locale/ILocale.java:33]: No, I think String is better here. IMO, Optional<Character> would be overengineering.

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java (hunk 1 above)
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java:23]: I don't see a problem with multi character separators, but I can add this for now.

Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java:23]: On the other hand, no, I would only do this if we really run into problems.

**Review by daniel-klose — APPROVED** [review]

**Pull request merged**